File: example/src/main/java/com/stripe/example/activity/LauncherActivity.java
package com.stripe.example.activity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.stripe.android.PaymentConfiguration;
import com.stripe.example.R;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class LauncherActivity extends AppCompatActivity {
    /*
     * Change this to your publishable key.
     *
     * You can get your key here: https://dashboard.stripe.com/account/apikeys
     */
    private static final String PUBLISHABLE_KEY =
            "put your key here";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_launcher);

        PaymentConfiguration.init(PUBLISHABLE_KEY);

        final RecyclerView examples = findViewById(R.id.examples);
        final LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
        linearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
        examples.setHasFixedSize(true);
        examples.setLayoutManager(linearLayoutManager);
        examples.setAdapter(new ExamplesAdapter(this));
    }

    private static final class ExamplesAdapter
            extends RecyclerView.Adapter<ExamplesAdapter.ExamplesViewHolder> {
        @NonNull private final Activity mActivity;
        @NonNull private final List<Item> mItems;

        private ExamplesAdapter(@NonNull Activity activity) {
            mActivity = activity;
            this.mItems = new ArrayList<>(Arrays.asList(
                    new Item(activity.getString(R.string.launch_payment_intent_example),
                            PaymentIntentActivity.class),
                    new Item(activity.getString(R.string.launch_setup_intent_example),
                            SetupIntentActivity.class),
                    new Item(activity.getString(R.string.payment_auth_example),
                            PaymentAuthActivity.class),
                    new Item(activity.getString(R.string.create_card_tokens),
                            PaymentActivity.class),
                    new Item(activity.getString(R.string.create_card_payment_methods),
                            PaymentMultilineActivity.class),
                    new Item(activity.getString(R.string.create_three_d_secure),
                            RedirectActivity.class),
                    new Item(activity.getString(R.string.launch_customer_session),
                            CustomerSessionActivity.class),
                    new Item(activity.getString(R.string.launch_payment_session),
                            PaymentSessionActivity.class),
                    new Item(activity.getString(R.string.launch_payment_session_from_fragment),
                            FragmentExampesActivity.class),
                    new Item(activity.getString(R.string.launch_pay_with_google),
                            PayWithGoogleActivity.class)
            ));
        }

        @NonNull
        @Override
        public ExamplesViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
            final View root = LayoutInflater.from(viewGroup.getContext())
                    .inflate(R.layout.launcher_item, viewGroup, false);
            return new ExamplesViewHolder(root);
        }

        @Override
        public void onBindViewHolder(@NonNull ExamplesViewHolder examplesViewHolder, int i) {
            ((TextView) examplesViewHolder.itemView).setText(mItems.get(i).mText);
            examplesViewHolder.itemView.setOnClickListener(v ->
                    mActivity.startActivity(new Intent(mActivity, mItems.get(i).mActivityClass)));
        }

        @Override
        public int getItemCount() {
            return mItems.size();
        }

        private static final class Item {
            @NonNull private final String mText;
            @NonNull private final Class<?> mActivityClass;

            private Item(@NonNull String text, @NonNull Class<?> activityClass) {
                this.mText = text;
                this.mActivityClass = activityClass;
            }
        }

        private static final class ExamplesViewHolder extends RecyclerView.ViewHolder {
            private ExamplesViewHolder(@NonNull View itemView) {
                super(itemView);
            }
        }
    }
}

File: example/src/main/java/com/stripe/example/activity/LauncherActivity.kt
package com.stripe.example.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.stripe.example.R
import com.stripe.example.databinding.LauncherActivityBinding

class LauncherActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val viewBinding = LauncherActivityBinding.inflate(layoutInflater)
        setContentView(viewBinding.root)

        val linearLayoutManager = LinearLayoutManager(this)
            .apply {
                orientation = LinearLayoutManager.VERTICAL
            }

        viewBinding.examples.run {
            setHasFixedSize(true)
            layoutManager = linearLayoutManager
            adapter = ExamplesAdapter(this@LauncherActivity)
        }
    }

    private class ExamplesAdapter constructor(
        private val activity: Activity
    ) : RecyclerView.Adapter<ExamplesAdapter.ExamplesViewHolder>() {
        private val items = listOf(
            Item(
                activity.getString(R.string.payment_auth_example),
                PaymentAuthActivity::class.java
            ),
            Item(
                activity.getString(R.string.card_brand_choice),
                CardBrandChoiceExampleActivity::class.java
            ),
            Item(
                activity.getString(R.string.create_card_token),
                CreateCardTokenActivity::class.java
            ),
            Item(
                activity.getString(R.string.create_card_payment_method),
                CreateCardPaymentMethodActivity::class.java
            ),
            Item(
                activity.getString(R.string.create_card_source),
                CreateCardSourceActivity::class.java
            ),
            Item(
                activity.getString(R.string.launch_customer_session),
                CustomerSessionActivity::class.java
            ),
            Item(
                activity.getString(R.string.launch_payment_session),
                PaymentSessionActivity::class.java
            ),
            Item(
                activity.getString(R.string.launch_payment_session_from_fragment),
                FragmentExamplesActivity::class.java
            ),
            Item(
                activity.getString(R.string.googlepaylauncher_example),
                GooglePayLauncherIntegrationActivity::class.java
            ),
            Item(
                activity.getString(R.string.googlepaycomposelauncher_example),
                GooglePayLauncherComposeActivity::class.java
            ),
            // This is for internal use so as not to confuse the user.
//            Item(
//                activity.getString(R.string.googlepayplayground_example),
//                GooglePayLauncherPlaygroundActivity::class.java
//            ),
            Item(
                activity.getString(R.string.googlepaypaymentmethodlauncher_example),
                GooglePayPaymentMethodLauncherIntegrationActivity::class.java
            ),
            Item(
                activity.getString(R.string.googlepaypaymentmethodcomposelauncher_example),
                GooglePayPaymentMethodLauncherComposeActivity::class.java
            ),
            Item(
                activity.getString(R.string.launch_confirm_pm_sepa_debit),
                ConfirmSepaDebitActivity::class.java
            ),
            Item(
                activity.getString(R.string.netbanking_payment_example),
                NetbankingListPaymentActivity::class.java
            ),
            Item(
                activity.getString(R.string.fpx_payment_example),
                FpxPaymentActivity::class.java
            ),
            Item(
                activity.getString(R.string.klarna_source_example),
                KlarnaSourceActivity::class.java
            ),
            Item(
                activity.getString(R.string.confirm_with_klarna),
                KlarnaPaymentActivity::class.java
            ),
            Item(
                activity.getString(R.string.confirm_with_affirm),
                AffirmPaymentActivity::class.java
            ),
            Item(
                activity.getString(R.string.confirm_with_alipay_native),
                AlipayPaymentNativeActivity::class.java
            ),
            Item(
                activity.getString(R.string.confirm_with_alipay_web),
                AlipayPaymentWebActivity::class.java
            ),
            Item(
                activity.getString(R.string.becs_debit_example),
                BecsDebitPaymentMethodActivity::class.java
            ),
            Item(
                activity.getString(R.string.bacs_debit_example),
                BacsDebitPaymentMethodActivity::class.java
            ),
            Item(
                activity.getString(R.string.multibanco_example),
                MultibancoActivity::class.java
            ),
            Item(
                activity.getString(R.string.sofort_example),
                SofortPaymentMethodActivity::class.java
            ),
            Item(
                "iDEAL Payment Example",
                IDEALPaymentMethodActivity::class.java
            ),
            Item(
                activity.getString(R.string.upi_example),
                UpiPaymentActivity::class.java
            ),
            Item(
                activity.getString(R.string.netbanking_example),
                NetbankingPaymentActivity::class.java
            ),
            Item(
                activity.getString(R.string.card_brands),
                CardBrandsActivity::class.java
            ),
            Item(
                activity.getString(R.string.simple_payment_method_example),
                SimpleConfirmationActivity::class.java
            ),
            Item(
                activity.getString(R.string.connect_example),
                ConnectExampleActivity::class.java
            ),
            Item(
                activity.getString(R.string.compose_example),
                ComposeExampleActivity::class.java
            ),
            Item(
                activity.getString(R.string.confirm_with_us_bank_account_entry_point),
                ConnectUSBankAccountActivity::class.java
            ),
            Item(
                activity.getString(R.string.manual_us_bank_account_example),
                ManualUSBankAccountPaymentMethodActivity::class.java
            ),
            Item(
                activity.getString(R.string.amazon_pay_example),
                AmazonPayActivity::class.java
            ),
            Item(
                activity.getString(R.string.cash_app_pay_example),
                CashAppPayActivity::class.java
            ),
            Item(
                "BLIK",
                BlikPaymentMethodActivity::class.java
            ),
            Item(
                activity.getString(R.string.revolut_pay_example),
                RevolutPayActivity::class.java
            ),
            Item(
                activity.getString(R.string.swish_example),
                SwishExampleActivity::class.java
            ),
            Item(
                activity.getString(R.string.mobilepay_example),
                MobilePayExampleActivity::class.java
            ),
            Item(
                activity.getString(R.string.alma_example),
                AlmaActivity::class.java
            ),
            Item(
                activity.getString(R.string.sunbit_example),
                SunbitActivity::class.java
            ),
            Item(
                activity.getString(R.string.billie_example),
                BillieActivity::class.java
            ),
            Item(
                activity.getString(R.string.satispay_example),
                SatispayActivity::class.java
            ),
            // This is for internal use so as not to confuse the user.
            Item(
                "StripeImage Example",
                StripeImageActivity::class.java
            ),
            Item(
                "Card Input Widget Compose Example",
                CardInputWidgetComposeExampleActivity::class.java
            )
        )

        override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ExamplesViewHolder {
            val root = activity.layoutInflater
                .inflate(R.layout.launcher_item, viewGroup, false)
            return ExamplesViewHolder(root)
        }

        override fun onBindViewHolder(examplesViewHolder: ExamplesViewHolder, i: Int) {
            val itemView = examplesViewHolder.itemView
            (itemView as TextView).text = items[i].text
            itemView.setOnClickListener {
                activity.startActivity(Intent(activity, items[i].activityClass))
            }
        }

        override fun getItemCount(): Int {
            return items.size
        }

        private data class Item constructor(val text: String, val activityClass: Class<*>)

        private class ExamplesViewHolder constructor(
            itemView: View
        ) : RecyclerView.ViewHolder(itemView)
    }
}
